profile-widget/diveeventitem.cpp

```diff
@@ -143,21 +158,30 @@ void DiveEventItem::setupPixmap()
 #undef EVENT_PIXMAP_BIGGER
 }
 
-void DiveEventItem::setupToolTipString()
+void DiveEventItem::setupToolTipString(struct gasmix *lastgasmix)
 {
 	// we display the event on screen - so translate
 	QString name = gettextFromC::instance()->tr(internalEvent->name);
 	int value = internalEvent->value;
 	int type = internalEvent->type;
 
 	if (event_is_gaschange(internalEvent)) {
+		struct icd_data icd_data;
 		struct gasmix *mix = get_gasmix_from_event(&displayed_dive, internalEvent);
 		name += ": ";
 		name += gasname(mix);
 
 		/* Do we have an explicit cylinder index?  Show it. */
 		if (internalEvent->gas.index >= 0)
 			name += tr(" (cyl. %1)").arg(internalEvent->gas.index + 1);
+		bool icd = isobaric_counterdiffusion(lastgasmix, mix, &icd_data);
+		if (icd_data.dHe < 0) {
+			if (icd)
+				name += tr("\nICD: ΔHe=%1% ΔN₂=%2%>%3%").arg(icd_data.dHe / 10).arg(icd_data.dN2 / 10).arg(-icd_data.dHe / 50);
+			else
+				name += tr("\nICD: ΔHe=%1% ΔN₂=%2%<%3%").arg(icd_data.dHe / 10).arg(icd_data.dN2 / 10).arg(-icd_data.dHe / 50);
+		}
```

Comment:
are the above arguments percentages - e.g. `%1%` says `25%` or should it just say `25`?

due to the duplication i would re-write the above if statement as:
```
if (icd_data.dHe < 0) {
	name += tr("\nICD: ΔHe=%1% ΔN₂=%2%%3%4%").arg(icd_data.dHe / 10).arg(icd_data.dN2 / 10)
		.arg(icd ? ">" : "<").arg(-icd_data.dHe / 50);
}
```
this reduces the translations strings and the LOC.

`>` and `<` are mathematical symbols and do not need translation. same goes for `=`.

Reply:
Yes, in fact I really would like to try to further improve the printout. At this moment I can also cover @neolit123 proposal.
But what really give to translation? The string in Lubomirs code example `"\nICD: ΔHe=%1% ΔN₂=%2%%3%4%"` is still very "ugly". Shall we maybe translate only "IDC", "He", "N₂" or "ΔHe" and "ΔN₂" or `"&#916;He"` and `"&#916;N&#8322;"` (used by Willem)?

Reply:
only `ICD` should be translated; the rest are either chemical symbols or math symbols. but the string loses context if you put only `ICD` for translation, thus my vote is to translate the whole string.